src/core/client/client.internal.ts

```diff
@@ -72,19 +73,29 @@ export class InternalClient {
   ): Promise<ProfileConfiguration> {
     const profileId = profileAstId(ast);
     if (this.superJson !== undefined) {
-      const profileSettings = this.superJson?.normalized.profiles[profileId];
+      const profileSettings = this.superJson.profiles[profileId];
       if (profileSettings === undefined) {
         throw profileNotInstalledError(profileId);
       }
 
-      // TODO: load priority and add it to ProfileConfiguration?
-      const priority = profileSettings.priority;
-      if (!priority.every(p => this.superJson?.normalized.providers[p])) {
-        throw unconfiguredProviderInPriorityError(
-          profileId,
-          priority,
-          Object.keys(this.superJson?.normalized.providers ?? [])
+      if ('file' in profileSettings) {
```

Comment:
I think we don't need `file` and exists check here - we already have profile AST.